rct229/rules/section5/section5rule36.py

```diff
@@ -0,0 +1,163 @@
+
+                return {
+                    "scc_dictionary_b": data["scc_dictionary_b"],
+                    "total_skylight_area_b": data["skylight_roof_areas_dictionary_b"][
+                        building_segment_b["id"]
+                    ]["total_skylight_area"],
+                    "total_skylight_area_p": data["skylight_roof_areas_dictionary_p"][
+                        building_segment_p["id"]
+                    ]["total_skylight_area"],
+                }
+
+            def create_context_list(self, context, data=None):
+                # create a list of regulated roof surfaces in baseline and proposed case
+                scc = data["scc_dictionary_b"]
+
+                surfaces_b = find_all("$..surfaces[*]", context.baseline)
+                surfaces_p = find_all("$..surfaces[*]", context.proposed)
+
+                # This assumes that the surfaces matched by IDs between proposed and baseline
+                matched_proposed_surfaces = match_lists_by_id(surfaces_b, surfaces_p)
+
+                proposed_baseline_surface_pairs = zip(
+                    surfaces_b, matched_proposed_surfaces
+                )
+
+                return [
+                    UserBaselineProposedVals(None, surface_b, surface_p)
+                    for surface_b, surface_p in proposed_baseline_surface_pairs
+                    if get_opaque_surface_type(surface_b) == OST.ROOF
+                    and scc[surface_b["id"]] != SCC.UNREGULATED
+                ]
+
+            class SurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule36.BuildingRule.BuildingSegmentRule.SurfaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    total_skylight_area_b = data["total_skylight_area_b"]
+                    total_skylight_area_p = data["total_skylight_area_p"]
+
+                    roof_b = context.baseline
+                    roof_p = context.proposed
+
+                    total_skylight_area_surface_b = sum(
+                        [
+                            subsurface.get("glazed_area", ZERO.AREA)
+                            + subsurface.get("opaque_area", ZERO.AREA)
+                            for subsurface in find_all("subsurfaces[*]", roof_b)
+                        ],
+                        ZERO.AREA,
+                    )
+                    total_skylight_area_surface_p = sum(
+                        [
+                            subsurface.get("glazed_area", ZERO.AREA)
+                            + subsurface.get("opaque_area", ZERO.AREA)
+                            for subsurface in find_all("subsurfaces[*]", roof_p)
+                        ],
+                        ZERO.AREA,
+                    )
+
```

Comment:
We have a divide-by-zero issue here.
I suggest:
1. Put `total_skylight_area_b` and `total_skylight_area_p` in a `get_calc_vals` method at the buliding_segment level.
2. At the roof level, put `total_skylight_area_surface_b` and `total_skylight_area_surface_p` in `calc_vals`.
3. In `rule_check`:
  - Handle the case where both `total_skylight_area_b` and `total_skylight_area_p` are zero (a passing case I think - they are trivially in the same ratios).
  - Handle the case where only one of these is zero (failing I would say - they can't be in the same ratios).
  - Finally check the ratios.

I would write this in a single return statement:
```
return (both segment areas are zero) or ((neither segment area is zero) and (the surface ratios are the same))
```

Reply:
Yes, you are right. My item 1. should have said to put the skylight area data in a create_data method.  Good catch.